kivy/core/window/_window_sdl2.pyx

```diff
@@ -745,13 +754,80 @@ cdef class _WindowSDL2Storage:
     def grab_mouse(self, grab):
         SDL_SetWindowGrab(self.win, SDL_TRUE if grab else SDL_FALSE)
 
+
+    def set_custom_titlebar(self, titlebar_widget):
+        SDL_SetWindowBordered(self.win, SDL_FALSE)
+        return SDL_SetWindowHitTest(self.win,custom_titlebar_handler_callback,<void *>titlebar_widget)
+
+    def hook_winProc(self):
+        IF UNAME_SYSNAME == 'Windows':
+            if platform == "win" and not old_windProc:
+    #             print("set hook")
+                global old_windProc
+                global new_windProc
+                new_windProc = WNDPROC(custom_wndProc)
+                old_windProc = SetWindowLong_WndProc_wrapper(self.get_window_info().window, new_windProc)
+
     @property
     def window_size(self):
         cdef int w, h
         SDL_GetWindowSize(self.win, &w, &h)
         return [w, h]
 
 
+def custom_wndProc(hwnd, msg, wParam, lParam):
+    IF UNAME_SYSNAME == 'Windows':
+        if msg == win32con.WM_NCCALCSIZE:
+            return 0
+        return ctypes.windll.user32.CallWindowProcW(old_windProc,
+                                                    hwnd, msg, wParam,
+                                                    lParam)
+def unbind_custom_wndProc(hwnd):
+    IF UNAME_SYSNAME == 'Windows':
+        global new_windProc, old_windProc
+        if new_windProc and old_windProc:
+            new_windProc = SetWindowLong_WndProc_wrapper(hwnd, old_windProc)
+
+cdef SDL_HitTestResult custom_titlebar_handler_callback(SDL_Window* win, const SDL_Point* pts, void* data) with gil:
+
+    # todo: discuss border size
+    cdef int border = 20 # pixels
+    cdef int w, h
+    SDL_GetWindowSize(<SDL_Window *> win, &w, &h)
+    # shift y origin in widget as sdl origin is in top-left
+
+    if pts.x < border and pts.y < border:
+        return SDL_HITTEST_RESIZE_TOPLEFT
+    elif pts.x < border < h - pts.y:
+        return SDL_HITTEST_RESIZE_LEFT
+    elif pts.x < border and h - pts.y < border:
+        return SDL_HITTEST_RESIZE_BOTTOMLEFT
+    elif w - pts.x < border > pts.y:
+        return SDL_HITTEST_RESIZE_TOPRIGHT
+    elif w - pts.x  > border > pts.y:
+        return SDL_HITTEST_RESIZE_TOP
+    elif w - pts.x  < border < h - pts.y:
+        return SDL_HITTEST_RESIZE_RIGHT
+    elif w - pts.x  < border > h - pts.y:
+        return SDL_HITTEST_RESIZE_BOTTOMRIGHT
+    elif w - pts.x  > border > h - pts.y:
+        return SDL_HITTEST_RESIZE_BOTTOM
+    widget = <object> data
+    in_drag_area = getattr(widget, 'in_drag_area', None)
+    if widget.collide_point(pts.x, h - pts.y):
+        if callable(in_drag_area):
+            if in_drag_area(pts.x, h - pts.y):
+                return SDL_HITTEST_DRAGGABLE
+            else:
+                return SDL_HitTestResult.SDL_HITTEST_NORMAL
+        for child in widget.walk():
+            drag = getattr(child, 'draggable', None)
```

Comment:
is the name property `draggable` ambiguous? should it be something like  `clickable` or `touchable` or something eles ?

Reply:
i fear it's a bit ambigious, yes, maybe `window_draggable`? or even something `custom_titlebar_draggable` ?

Reply:
no, see the code will return `None` if it child does not have the property, so if we use  `window_draggable` the user will need to set it to false for it to become clickable 
as the conditoin for being clickable is
```py
            if drag is not None and not drag and child.collide_point(pts.x, h - pts.y):
                return SDL_HitTestResult.SDL_HITTEST_NORMAL
```

Reply:
instead if we use a property like `clickable = True` , i think it is easier to remember,
instead of `draggable = false `